### PR TITLE
Add sensitive support for db password and ldap password

### LINKED ISSUE #482: Add sensitive option for database_config
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 6.25.1
- Ruby: 2.5.9p229
- Distribution: RHEL 8
- Module version: v6.0.0

## How to reproduce (e.g Puppet code you use)

database_config         => {
      'type'            => 'postgresql',
      'url'             => "jdbc:postgresql://${db_host}/${_servername}",
      'username'        => $db_username,
      'password'        => $db_password.unwrap,
      'driverClassName' => 'org.postgresql.Driver',
    },

## What are you seeing
Notice: /Stage[main]/Rundeck::Config::Global::Rundeck_config/File[/etc/rundeck/rundeck-config.groovy]/content:
--- /etc/rundeck/rundeck-config.groovy  2022-01-11 09:02:27.648577738 +0100
+++ /tmp/puppet-file20220111-42398-1beilhv      2022-01-11 09:06:00.912297639 +0100
@@ -12,7 +12,7 @@
   url = "jdbc:postgresql://localhost/rundeck"
   driverClassName = "org.postgresql.Driver"
   username = "my_user"
'-'  password = "my_pass"
'+'  password = "my_password"
   dialect = ""
 }

## What behaviour did you expect instead
Notice: /Stage[main]/Rundeck::Config::Global::Rundeck_config/File[/etc/rundeck/rundeck-config.groovy]/content: value [redacted]

### PR DESCRIPTION
**Pull request by nosrio**
#### Pull Request (PR) description
This PR add sensitive support for database password and ldap password.

#### This Pull Request (PR) fixes the following issues
Fixes #482

### CODE REVIEW

**Comment by kpolucas** [comment]
+1

File: templates/_auth_ldap.epp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  providerUrl="<%= $rundeck::config::auth_config['ldap']['url'] %>"
+<%-} else {-%>
+  providerUrl="ldap://<%= $rundeck::config::auth_config['ldap']['server'] %>:<%= $rundeck::config::auth_config['ldap']['port'] %>"
+<%-}-%>  
+  authenticationMethod="simple"
+  forceBindingLogin="<%= $rundeck::config::auth_config['ldap']['force_binding'] %>"
+<%- if $rundeck::config::auth_config['ldap']['force_binding_use_root'] != undef {-%>
+  forceBindingLoginUseRootContextForRoles="<%= $rundeck::config::auth_config['ldap']['force_binding_use_root'] %>"
+<%- } -%>
+<%- if $rundeck::config::auth_config['ldap']['bind_dn'] != undef {-%>
```
Comment (bastelfreak) [inline templates/_auth_ldap.epp:14]: you can simplify those if conditions a bit:
```suggestion
<%- if $rundeck::config::auth_config['ldap']['bind_dn'] {-%>
```

**Comment by nosrio** [comment]
Done @bastelfreak . I also remove it from other places.

**Comment by CobreQ** [comment]
+1

**Comment by leandrobalma** [comment]
+1

**Comment by nosrio** [comment]
@bastelfreak are you planning on tagging a new version?

**Comment by bastelfreak** [comment]
yes, as soon as I find some spare time.

**Pull request merged**